packages/textcomplete-core/src/SearchResult.ts

```diff
@@ -10,7 +10,11 @@ export class SearchResult<T = unknown> {
     private readonly strategy: Strategy<T>
   ) {}
 
-  replace(beforeCursor: string, afterCursor: string): [string, string] | void {
+  getReplacementData(beforeCursor: string, afterCursor: string): {
+    start: number
+    end: number
+    text: string
+  } | void {
```

Comment:
nits, IMO: Although `null` type is more commonly used in order to express a empty value, I think this code is ok because the original code uses `void` type to express a empty value.

Reply:
I changed this because I thought it would be more appropriate to return null.